lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -890,7 +887,14 @@ private function persistNew($class, $entity)
             $idValue = $idGen->generate($this->em, $entity);
 
             if ( ! $idGen instanceof \Doctrine\ORM\Id\AssignedGenerator) {
-                $idValue = array($class->identifier[0] => $idValue);
+                $idField = $class->identifier[0];
```

Comment:
But then it's a bit confusing... should this conversion be generators' responsibility?